.github/workflows/build.yml

```
@@ -1,41 +1,57 @@
 name: build
 
 on:
+  workflow_dispatch:
   push:
     branches: [ master ]
   pull_request:
     branches: [ master ]
 
 jobs:
-  build:
-
+  check:
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9, '3.10']
-
+    
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
-      with:
-        python-version: ${{ matrix.python-version }}
+    - uses: actions/checkout@v3
+
     - name: Black Code Formatter
       uses: lgeiger/black-action@v1.0.1
       with:
         args: ". --check"
+
     - name: Mypy Check
       uses: jpetrucciani/mypy-check@0.761
       with:
         path: 'src'
+    
+  build:
+    needs: [check]
+    strategy:
+      fail-fast: false
+      matrix:
+        operating-system: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: [3.6, 3.7, 3.8, 3.9, '3.10', '3.11-dev']
```

I propose to remove `3.11-dev` because there is an issue in the corresponding pr #17 , something is wrong with python 3.11-dev and poetry, and is not resolved yet.

```suggestion
        python-version: [3.6, 3.7, 3.8, 3.9, '3.10']
```